addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java

```diff
@@ -0,0 +1,865 @@
+                        // TODO show output that Method purging didn't work.
+                    } else {
+                        /* Now check If we have a hit to determine that the response is from cache. */
+                        httpSender.sendAndReceive(msg);
+                        if (msg.getResponseHeader().getStatusCode()
+                                == base.getResponseHeader().getStatusCode()) {
+                            indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
+                            if (this.checkCacheHit(indicValue, cache)) {
+                                cache.setCacheBusterFound(true);
+                                cache.setCacheBusterIsHttpMethod(true);
+                                cache.setCacheBusterName(httpMethods[i]);
+                                this.bustedMessage = msg;
+                            } else {
+                                // TODO show output that Method purging didn't work.
+                            }
+                        }
+                    }
+                }
+            } else {
+                /* Since no indicator was found we try time difference. */
+                /* Decide threshold */
+                if (config.getCacheBustingThreshold() == -1) {
+                    HttpRequestHeader headers2 = new HttpRequestHeader();
+                    List<Integer> times = new ArrayList<Integer>();
+                    for (int j = 0; j < 2; j++) {
+                        headers2.setURI(new URI(url, true));
+                        headers2.setVersion(HttpHeader.HTTP11);
+                        if (j % 2 == 0) {
+                            headers2.setMethod(httpMethods[i]);
+                        } else {
+                            switch (method) {
+                                case GET:
+                                    headers2.setMethod(HttpRequestHeader.GET);
+                                    break;
+                                case POST:
+                                    headers2.setMethod(HttpRequestHeader.POST);
+                                    break;
+                                default:
+                                    throw new IllegalArgumentException(
+                                            Constant.messages.getString(
+                                                    METHOD_NOT_SUPPORTED, method));
+                            }
+                        }
+                        HttpMessage msg = new HttpMessage();
+                        msg.setRequestHeader(headers2);
+                        httpSender.sendAndReceive(msg);
+
+                        if (msg.getResponseHeader().getStatusCode()
+                                == base.getResponseHeader().getStatusCode()) {
+                            times.add(msg.getTimeElapsedMillis());
+                        }
+                    }
+                    if (times.get(0) > times.get(1)) {
```

Comment:
Okay so one thing I've found running `shouldFindCacheWithHeaderCacheBusterWhenNoIndicatorPresent` individually locally and repeatedly/serially is that times is sometimes only a single element here, so it will sometimes throw and out of bounds exception. This condition should likely be `if (times.size() > 1 && times.get(0) > times.get(1)) {`